userapi/storage/accounts/postgres/profile_table.go

```diff
@@ -105,3 +113,21 @@ func (s *profilesStatements) setDisplayName(
 	_, err = s.setDisplayNameStmt.ExecContext(ctx, displayName, localpart)
 	return
 }
+
+func (s *profilesStatements) selectProfilesBySearch(
+	ctx context.Context, searchString string, limit int,
+) ([]authtypes.Profile, error) {
+	var profiles []authtypes.Profile
+	rows, err := s.selectProfilesBySearchStmt.QueryContext(ctx, fmt.Sprintf("%%%s%%", searchString), limit)
```

Comment:
I don't understand how `%%%s%%` works. It's `%% %s %%` right? Why double `%`?

Reply:
I've added comments to this, but it's pretty much because we're giving this parameter to `LIKE $1` in the SQL statement and we want to wildcard-search. `%%` escapes the `%` character, so it's basically turning into `LIKE %searchString%`.